package/gluon-core/luasrc/lib/gluon/upgrade/210-interface-mesh

```diff
@@ -4,35 +4,51 @@ local site = require 'gluon.site'
 local uci = require('simple-uci').cursor()
 local util = require 'gluon.util'
 
-local mesh_interfaces = util.get_role_interfaces(uci, 'mesh')
-local uplink_interfaces = util.get_role_interfaces(uci, 'uplink')
+local mesh_interfaces = util.get_role_interfaces_with_options(uci, 'mesh', {'batadv_hop_penalty'})
+local uplink_interfaces = util.get_role_interfaces_with_options(uci, 'uplink', {})
 
 local mesh_interfaces_uplink = {}
 local mesh_interfaces_other = {}
-for _, iface in ipairs(mesh_interfaces) do
-	if util.contains(uplink_interfaces, iface) then
-		table.insert(mesh_interfaces_uplink, iface)
+for iface, options in pairs(mesh_interfaces) do
+	if uplink_interfaces[iface] ~= nil then
+		table.insert(mesh_interfaces_uplink, {iface = iface, options = options})
 	else
-		table.insert(mesh_interfaces_other, iface)
+		table.insert(mesh_interfaces_other, {iface = iface, options = options})
 	end
 end
 
+local function extract_ifaces(interfaces)
+	local ifnames = {}
+	local max_hop_penalty = 0
+	for _, iface in ipairs(interfaces) do
+		table.insert(ifnames, iface['iface'])
+		local options = iface['options']
+		if options['batadv_hop_penalty'] and tonumber(options['batadv_hop_penalty']) > max_hop_penalty then
+			max_hop_penalty = options['batadv_hop_penalty']
+		end
```

Comment:
You only `tonumber()` the hop penalty for the comparison, but the value assigned to `max_hop_penalty` must be converted as well. I'd write the code like this:

```suggestion
		table.insert(ifnames, iface.iface)
		let hop_penalty = tonumber(iface.options.batadv_hop_penalty) or 0
		if max_hop_penalty < hop_penalty  then
			max_hop_penalty = hop_penalty
		end
```
(untested)